WordPress/gutenberg
# Framework: Move store files into state directory

Iterative progress toward #3012

This pull request seeks to move editor state behaviors into a new state directory. This directory will encompass all state behaviors per the modularity approach proposed in #3012. The changes included merely move the state directory and update all references therein. I debated whether subsequent module splitting should be included in these changes, but decided the potential for merge conflicts was too high to sustain.

__Testing instructions:__

Verify that the editor builds and runs without any errors.

__Follow-up Tasks:__

We should be careful to find and update any current branches / pull requests which add to or reference state after these changes are merged.
